runtime/compiler/runtime/JitRuntime.cpp

```diff
@@ -645,6 +645,14 @@ void J9FASTCALL _jitProfileStringValue(uintptr_t value, int32_t charsOffset, int
       {
       readValues = true;
 
+      uintptr_t startOfData = value;
+#if defined(J9VM_GC_ENABLE_SPARSE_HEAP_ALLOCATION)
+      if (TR::Compiler->om.isOffHeapAllocationEnabled())
+         {
+         startOfData = *((uintptr_t *) (value + TR::Compiler->om.offsetOfContiguousDataAddrField()));
```

Comment:
offsetOfContiguousDataAddrField is still within the header size, not at the start of data yet.  if "value" is an off-heap array (admittedly unlikely), you cannot get to the start of data this way without loading dataAddr.  This looks incorrect.

Reply:
We're dereferencing the sum of value + offsetOfContiguousDataAddrField to get the address of the start of data, or the dereference is incorrect?

Reply:
ah ... didn't notice the dereferencing